### DISCUSSION TITLE
Repo, Branch, Fork Confusion

### DISCUSSION THREAD

**Discussion by Crip-Zone** [community/community#58338] [New to GitHub] [answered]
Hi,

I've created a repo for Simple Machines Forums (SMF) which is the various skins/looks of the forum...for those who don't know. I currently have 3 repos, each named after 3 themes we've updated for SMF version 2.1.3 and each is currently named with the version number after it, (Ambassador-2.1.3.c, for example. However, SMF has just released version 2.1.4.

My problem is that how should I have things set up so that I can incorporate new theme versions and updates without creating new repos, if possible. Or what would be the best way to do this? I just started this recently an am trying to not have to redo everything later after doing a lot of work. LOL Do I need forks, branches or what? If so, how and in what order?

Thanks in advance,
Steve

**Comment by airtower-luna** [comment, marked as answer]
If you want to mark versions, in Git *tags* are generally the right thing. A tag marks a specific commit (e.g. "this commit is version 1.2.3"), see [Git Basics - Tagging](https://git-scm.com/book/en/v2/Git-Basics-Tagging). GitHub additionally has *releases*, which are based on tags and add some more metadata (and optionally files).

If you want to maintain multiple stable versions over time, having a branch for each stable series (e.g. `2.x`) is probably good (in addition to tags, for the actual maintenance work).

**Reply by Crip-Zone** [reply]
Thank you very much! :)
